### PR TITLE
Setup express app

### PR DESCRIPTION
**Pull request by wSedlacek**
# Description

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Comment by wSedlacek** [comment]
[[Developer] - Setup repository file structure and extension config](https://trello.com/c/ZTpVL88l/25-developer-setup-repository-file-structure-and-extension-config)

File: src/models/Middleware.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import * as express from 'express';
+
+export type Middleware<T = {}> = (
```
Comment (EricBanker12) [inline src/models/Middleware.ts]: I'm confused by what is going on here. A comment would help

File: src/models/Middleware.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    options?: T
+) => (req: express.Request, res: express.Response, next: express.NextFunction) => void;
```
Comment (EricBanker12) [inline src/models/Middleware.ts:8]: I'm confused by what is going on here. A comment would help

**Review by EricBanker12 — APPROVED** [review]
I'd like a comment on `src/model/Middleware.ts`, but otherwise it looks good after adding testing.

File: src/models/Middleware.ts (hunk 2 above)
Comment (wSedlacek) [reply src/models/Middleware.ts:8]: Added a comment to clarify on this type definition, let me know if this clears things up.

**Pull request merged**